cpp/test/sg/fil_test.cu

```diff
@@ -886,4 +889,139 @@ TEST_P(TreeliteThrowSparse8FilTest, Import) { check(); }
 INSTANTIATE_TEST_CASE_P(FilTests, TreeliteThrowSparse8FilTest,
                         testing::ValuesIn(import_throw_sparse8_inputs));
 
+template <typename T>
+__device__ void serial_multi_reduction(T* in, T* out, int set_size,
+                                       int n_sets) {
+  __syncthreads();
+  if (threadIdx.x < set_size) {
+    int reduction_id = threadIdx.x;
+    T sum = 0;
+    for (int set = 0; set < n_sets; ++set)
+      sum += in[reduction_id + set * set_size];
+    out[reduction_id] = sum;
+  }
+  __syncthreads();
+}
+
+// the most threads a block can have
+const int max_threads = 1024;
+
+struct MultiReductionTestParams {
+  int radix, set_size, n_sets;
```

Comment:
It's also better to define each parameter on its own line, and have a comment describing it.